test/unit/org/apache/cassandra/db/commitlog/CommitLogBytemanTest.java

```diff
@@ -0,0 +1,171 @@
+    protected static final String STANDARD = "Standard";
+    private static final String CUSTOM = "Custom";
+
+    private static JVMStabilityInspector.Killer oldKiller;
+    private static KillerForTests testKiller;
+
+    public static volatile boolean failSync = false;
+
+    @BeforeClass
+    public static void beforeClass() throws ConfigurationException
+    {
+        // Disable durable writes for system keyspaces to prevent system mutations, e.g. sstable_activity,
+        // to end up in CL segments and cause unexpected results in this test wrt counting CL segments,
+        // see CASSANDRA-12854
+        org.apache.cassandra.db.commitlog.CommitLogBytemanTest.failSync = false;
+        KeyspaceParams.DEFAULT_LOCAL_DURABLE_WRITES = false;
+
+        SchemaLoader.prepareServer();
+        StorageService.instance.getTokenMetadata().updateHostId(UUID.randomUUID(), FBUtilities.getBroadcastAddressAndPort());
+
+        MemtableParams skipListMemtable = MemtableParams.fromMap(ImmutableMap.of("class", "SkipListMemtable"));
+
+        TableMetadata.Builder custom =
+                TableMetadata.builder(KEYSPACE, CUSTOM)
+                        .addPartitionKeyColumn("k", IntegerType.instance)
+                        .addClusteringColumn("c1", MapType.getInstance(UTF8Type.instance, UTF8Type.instance, false))
+                        .addClusteringColumn("c2", SetType.getInstance(UTF8Type.instance, false))
+                        .addStaticColumn("s", IntegerType.instance)
+                        .memtable(skipListMemtable);
+
+        SchemaLoader.createKeyspace(KEYSPACE,
+                KeyspaceParams.simple(1),
+                SchemaLoader.standardCFMD(KEYSPACE, STANDARD, 0, AsciiType.instance, BytesType.instance).memtable(skipListMemtable),
+                custom);
+        CompactionManager.instance.disableAutoCompaction();
+
+        testKiller = new KillerForTests();
+
+        // While we don't want the JVM to be nuked from under us on a test failure, we DO want some indication of
+        // an error. If we hit a "Kill the JVM" condition while working with the CL when we don't expect it, an aggressive
+        // KillerForTests will assertion out on us.
+        oldKiller = JVMStabilityInspector.replaceKiller(testKiller);
+    }
+
+    @AfterClass
+    public static void afterClass()
+    {
+        JVMStabilityInspector.replaceKiller(oldKiller);
+    }
+
+    @Before
+    public void beforeTest() throws IOException
+    {
+        CommitLog.instance.resetUnsafe(true);
+    }
+
+    @After
+    public void afterTest()
+    {
+        testKiller.reset();
+    }
+
+    @Test
+    @BMRules(rules = { @BMRule(name = "Fail sync in CommitLog",
+            targetClass = "CommitLog",
+            targetMethod = "sync",
+            condition = "org.apache.cassandra.db.commitlog.CommitLogBytemanTest.failSync",
+            action = "throw new java.lang.RuntimeException();") } )
```

Comment:
is it possible to add exception message? so that we know for sure it's a test exception when we see it (in the logs or wherever)